pyxem/utils/peakfinders2D.py

```diff
@@ -340,7 +341,9 @@ def find_peaks_dog(
     sensitive to fluctuations in intensity near the edges of the image.
 
     """
-    z = z / np.max(z)
+    z = z.astype(np.float32)
```

Comment:
What's the reason for this?

Reply:
For some of my data, if the dtype was `uint16` the peak finding would not work at all. So this was a (fairly) dirty way of solving that issue.

I'll go back and see if I can make a minimal working example of this.

Reply:
Perhaps make casting optional, in the same way as `normalize`?

Reply:
e.g. extreme example what if z.dtype is float64 and you're downcasting? :)

Reply:
This replicates the bug, with skimage 0.16.2:

```python
import numpy as np
from skimage.feature import blob_dog

def find_peaks_dog(
    z, min_sigma=1.0, max_sigma=50.0, sigma_ratio=1.6, threshold=0.2, overlap=0.5):
    blobs = blob_dog(
        z, min_sigma=min_sigma, max_sigma=max_sigma,
        sigma_ratio=sigma_ratio, threshold=threshold,
        overlap=overlap)
    centers = blobs[:, :2]
    return centers

data = np.zeros((20, 20), dtype=np.uint16)
data[7:13, 7:13] = 10

print("dtype uint16: ", find_peaks_dog(data))
print("dtype float32: ", find_peaks_dog(data.astype('float32')))
```
This gives: 
```python
dtype uint16:  []
dtype float32:  [[9. 9.]]
```
I'm guessing previously in the old implementation, the normalization `z = z / np.max(z)`, returns a float. So the change of dtype has been there the whole time, but "hidden" inside the normalization. In fact, it seems like doing this returns a `float64`, which can probably be optimized for better performance.

Reply:
Good catch